File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/java_solutions/529-minesweeper.java
class Solution {
    public int[] dx = {-1, -1, -1, 0, 0, 1, 1, 1};
    public int[] dy = {-1, 0, 1, -1, 1, -1, 0, 1};

    public char[][] updateBoard(char[][] board, int[] click) {
        int x = click[0], y = click[1];
        if (board[x][y] == 'M') {
            board[x][y] = 'X';
        } else {
            dfs(board, x, y);
        }
        return board;
    }

    public void dfs(char[][] board, int x, int y) {
        if (x < 0 || x >= board.length || y < 0 || y >= board[0].length || board[x][y] != 'E') {
            return;
        }
        int count = 0;
        for (int i = 0; i < 8; i++) {
            int nx = x + dx[i];
            int ny = y + dy[i];
            if (nx >= 0 && nx < board.length && ny >= 0 && ny < board[0].length && board[nx][ny] == 'M') {
                count++;
            }
        }
        if (count > 0) {
            board[x][y] = (char) ('0' + count);
        } else {
            board[x][y] = 'B';
            for (int i = 0; i < 8; i++) {
                int nx = x + dx[i];
                int ny = y + dy[i];
                dfs(board, nx, ny);
            }
        }
    }
}

File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/cpp_solutions/529-minesweeper.cpp
class Solution {
public:
    vector<vector<char>> updateBoard(vector<vector<char>>& board, vector<int>& click) {
        int x = click[0], y = click[1];
        if (board[x][y] == 'M') {
            board[x][y] = 'X';
            return board;
        }
        dfs(board, x, y);
        return board;
    }

private:
    void dfs(vector<vector<char>>& board, int x, int y) {
        if (x < 0 || x >= board.size() || y < 0 || y >= board[0].size() || board[x][y] != 'E') {
            return;
        }
        int mines = countMines(board, x, y);
        if (mines > 0) {
            board[x][y] = '0' + mines;
            return;
        }
        board[x][y] = 'B';
        for (int i = -1; i <= 1; ++i) {
            for (int j = -1; j <= 1; ++j) {
                if (i == 0 && j == 0) continue;
                dfs(board, x + i, y + j);
            }
        }
    }

    int countMines(vector<vector<char>>& board, int x, int y) {
        int count = 0;
        for (int i = -1; i <= 1; ++i) {
            for (int j = -1; j <= 1; ++j) {
                if (i == 0 && j == 0) continue;
                int nx = x + i, ny = y + j;
                if (nx >= 0 && nx < board.size() && ny >= 0 && ny < board[0].size() && board[nx][ny] == 'M') {
                    ++count;
                }
            }
        }
        return count;
    }
};